### PR TITLE
Use std::next in LabelMap, and other style improvements

### PR DESCRIPTION
**Pull request by N-Dekker**
LabelMap code clean-up, inspired by LabelMap code changes from pull request #4400

### CODE REVIEW

File: Modules/Filtering/LabelMap/include/itkLabelMap.hxx [hunk 1]
```diff
@@ -183,39 +175,25 @@ template <typename TLabelObject>
...
-      return it->second;
-    }
-    ++i;
+    itkExceptionMacro("Can't access to label object at position "
+                      << pos << ". The label map has only " << numberOfLabelObjects << " label objects registered.");
   }
-  itkExceptionMacro("Can't access to label object at position " << pos << ". The label map has only "
-                                                                << this->GetNumberOfLabelObjects()
-                                                                << " label objects registered.");
+  return std::next(m_LabelObjectContainer.cbegin(); pos);
```
Comment (issakomi) [inline Modules/Filtering/LabelMap/include/itkLabelMap.hxx:183]: typo

Comment (N-Dekker) [reply Modules/Filtering/LabelMap/include/itkLabelMap.hxx:183]: @issakomi Thanks but do you mean the same type that @dzenanz mentioned?

Comment (issakomi) [reply Modules/Filtering/LabelMap/include/itkLabelMap.hxx:183]: I mean `;` after ` cbegin()`

File: Modules/Filtering/LabelMap/include/itkLabelMap.hxx [hunk 2]
```diff
@@ -183,39 +183,25 @@ template <typename TLabelObject>
...
-
-  for (auto it = m_LabelObjectContainer.begin(); it != m_LabelObjectContainer.end(); ++it)
+  if (const auto numberOfLabelObjects = m_LabelObjectContainer.size(); numberOfLabelObjects <= pos)
   {
-    if (i == pos)
-    {
-      return it->second;
-    }
-    ++i;
+    itkExceptionMacro("Can't access to label object at position "
```
[the remark this answers is not in the document]
Comment (N-Dekker) [reply Modules/Filtering/LabelMap/include/itkLabelMap.hxx]: Thanks @dzenanz, addressed by the force-push that I just did!

File: Modules/Filtering/LabelMap/include/itkLabelMap.hxx (hunk 1 above)
Comment (N-Dekker) [reply Modules/Filtering/LabelMap/include/itkLabelMap.hxx]: Thanks @issakomi, also addressed by the force-push that I just did!

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
:1st_place_medal: lovely to have this standard library iterator support and to be able to use this syntax and interfaces!

**Pull request merged**